src/parser.ts

```diff
@@ -15,13 +15,119 @@ type TokenKind =
   | 'WHITESPACE'
   | 'EOF';
 
+/**
+ * Parses a given string, returning a Tree object.
+ *
+ * @param input
+ * @returns Resulting tree.
+ */
+function parse(input: string) {
```

Comment:
I would also return the type (`ChomTree`).

This actually goes for all of your functions.